### DISCUSSION TITLE
Caption for tablex

### DISCUSSION THREAD

**Discussion by yoonghm** [PgBiel/typst-tablex#88] [Q&A] [answered]
How do you put a caption for a tablex. I use the following code for standard table:

	//Insert a table, embedded within a figure for caption
	#figure(
		table(
			columns: (1fr, auto, auto),
			inset: 10pt,
			align: horizon,
			[   ],
			[  *Area*  ],
			[  *Parameters*  ],
			image("./assets/aws.svg"),
			$ pi h (D^2 - d^2) / 4 $,
			[
				$h$: height \
				$D$: outer radius \
				$d$: inner radius
			],
			image("./assets/aws.svg"),
			$ sqrt(2) / 12 a^3 $,
			[  $a$: edge length  ],
		),
		caption: [A sample table]
	)

**Comment by PgBiel** [comment, marked as answer]
Hi, you have to add `kind: table` to your figure so that it recognizes the tablex output as a table. See also #7

```typ
//Insert a table, embedded within a figure for caption
#figure(
	tablex(
		columns: (1fr, auto, auto),
		inset: 10pt,
		align: horizon,
		[   ],
		[  *Area*  ],
		[  *Parameters*  ],
		image("./assets/aws.svg"),
		$ pi h (D^2 - d^2) / 4 $,
		[
			$h$: height \
			$D$: outer radius \
			$d$: inner radius
		],
		image("./assets/aws.svg"),
		$ sqrt(2) / 12 a^3 $,
		[  $a$: edge length  ],
	),
        kind: table,
	caption: [A sample table]
)
```

**Reply by yoonghm** [reply]
Thanks
